roles/core/dns_server/tasks/main.yml

```diff
@@ -1,8 +1,29 @@
 ---
+- name: Gathering OS specific variables
+  # This task gathers variables defined in OS specific files.
+  #
+  # Search vars in:
+  #  - <distribution>_<major>.yml    # eg. CentOS_8.yml
+  #  - <os_family>_<major>.yml       # eg. RedHat_8.yml
+  #  - <distribution>.yml            # eg. CentOS.yml
+  #  - <os_family>.yml               # eg. RedHat.yml
+  #
+  # If no OS specific file is found, the role will default to vars/main.yml
+  #
+  include_vars: "{{ item }}"
+  with_first_found:
+    - files:
+        - "vars/{{ ansible_facts.distribution | replace(' ','_') }}_{{ ansible_facts.distribution_major_version }}.yml"
+        - "vars/{{ ansible_facts.os_family }}_{{ ansible_facts.distribution_major_version }}.yml"
+        - "vars/{{ ansible_facts.distribution | replace(' ','_') }}.yml"
+        - "vars/{{ ansible_facts.os_family }}.yml"
+      skip: True
+
 - name: Install packages
   package:
-    name: "{{ packages_to_install[(ansible_distribution|lower|replace(' ','_'))]['_'+ansible_distribution_major_version] }}"
+    name: "{{ item }}"
     state: present
+  loop: "{{ packages_to_install }}"
```

Comment:
looping over package isn't really recommended, can you supply the package list in "name" without looping ?

From Ansible loops documentation :
You can pass a list directly to a parameter for some plugins. Most of the packaging modules, like yum – Manages packages with the yum package manager and apt – Manages apt-packages, have this capability. When available, passing the list to a parameter is better than looping over the task.

Reply:
Sure. Thanks.

Reply:
Done. I did not know we could pass a list to package. Thanks for the info.